beginend.el

```diff
@@ -49,25 +49,32 @@
   (re-search-backward "[^[:space:]]" nil t)
   (forward-char))
 
+(defmacro beginend--double-tap (extremum &rest body)
+  "Go to point EXTREMUM if executing BODY did not change point."
+  (declare (debug (form body))
+           (indent 1))
+  (let ((oldpos-var (make-symbol "old-position"))
+        (extremum-var (make-symbol "extremum")))
+    `(let ((,oldpos-var (point))
+           (,extremum-var ,extremum))
+       (goto-char ,extremum-var)
+       ,@body
+       (if (equal ,oldpos-var (point))
+           (goto-char ,extremum-var)
+         (unless (eq ,oldpos-var ,extremum-var)
```

Comment:
Use `=` for numbers.  Maybe change to `when /=`, I'm not sure which reads better, might be too subjective.